### PR TITLE
8264051: Remove unused TRAPS parameters from runtime functions

### PR DESCRIPTION
**Pull request by coleenp**
This change removes the TRAPS parameter from compute_modifier_flags(), lookup_instance_method_in_klasses and nest_host_error.

There's a progressive effort to remove cases where the last parameter of a function is THREAD, and it's unclear why it is ignoring an exception or whether an exception is expected, if it doesn't subsequently have a check for HAS_PENDING_EXCEPTION.

Tested locally with tier1 tests and tier1 tests on 4 Oracle platforms in progress.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264051](https://bugs.openjdk.java.net/browse/JDK-8264051): Remove unused TRAPS parameters from runtime functions

### Reviewers
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**) ⚠️ Review applies to 57f5254eb67278fa64dd2267d48ce4505c359da1
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/jdk pull/3157/head:pull/3157`
`$ git checkout pull/3157`

To update a local copy of the PR:
`$ git checkout pull/3157`
`$ git pull https://git.openjdk.java.net/jdk pull/3157/head`

### CODE REVIEW

**Review by iklam — APPROVED** [review]
LGTM. Thanks for doing the cleanup.

File: src/hotspot/share/oops/instanceKlass.hpp [hunk 1]
```diff
@@ -465,7 +465,7 @@ class InstanceKlass: public Klass {
   jushort nest_host_index() const { return _nest_host_index; }
   void set_nest_host_index(u2 i)  { _nest_host_index = i; }
   // dynamic nest member support
-  void set_nest_host(InstanceKlass* host, TRAPS);
+  void set_nest_host(InstanceKlass* host, JavaThread* current);
```
Comment (dholmes-ora) [inline src/hotspot/share/oops/instanceKlass.hpp:468]: I think we should drop the thread parameter for this method and nest_host_error and avoid the tramp data. We only need the current thread for a ResourceMark in a logging clause in set_nest_host, and for a ConstantPoolHandle in nest_host_error. Neither of these codepaths are hot so we can just manifest Thread::current() when needed there.
Then you can also remove the thread parameter from print_nest_host_error_on.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Nice cleanup Coleen!

I have a further suggested change to simplify things further - see below.

Thanks,
David

File: src/hotspot/share/oops/instanceKlass.hpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/share/oops/instanceKlass.hpp:468]: You're right. I can remove a lot of thread arguments this way.

Comment (coleenp) [reply src/hotspot/share/oops/instanceKlass.hpp]: Done.  Reran runtime/Nestmates tests and tier1.

**Review by dholmes-ora — APPROVED** [review]
Looks good!

Thanks,
David

**Comment by coleenp** [comment]
Thanks for reviewing!
/integrate

**Pull request closed without merging**